devel-common/src/tests_common/test_utils/fernet.py

```diff
@@ -0,0 +1,48 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from pathlib import Path
+
+import yaml
+from cryptography.fernet import Fernet
+
+
+def generate_fernet_key_string() -> str:
+    """Generate a new Fernet key."""
+    return Fernet.generate_key().decode()
+
+
+def update_environment_variable_from_compose_yaml(file_path: str | Path) -> None:
```

Comment:
One comment here - I somehow do not like the idea of dynamically generating the compose file, that sounds like something rather difficult to debug. While it's ok to do it now, I feel that better solution would be to read the key from a fernet key passed as a variable to the compose file. This can be done  - for example by having several compose files combined together (you just pass those files as array of compse files). I will submit my change soon for task-sdk testing to show how I have done that to make testing easier (once I solve one last problem).